pulpcore/cli/ansible/remote.py

```diff
@@ -0,0 +1,232 @@
+        group.commands.pop("update_collection", None)
+    else:
+        group.commands.pop("create_role", None)
+        group.commands.pop("update_role", None)
+    return value
+
+
+@click.group()
+@click.option(
+    "-t",
+    "--type",
+    "remote_type",
+    type=click.Choice(["role", "collection"], case_sensitive=False),
+    default="role",
+    callback=_type_callback,
+    is_eager=True,
+)
+@pass_pulp_context
+@click.pass_context
+def remote(ctx: click.Context, pulp_ctx: PulpContext, remote_type: str) -> None:
+    if remote_type == "role":
+        ctx.obj = PulpAnsibleRoleRemoteContext(pulp_ctx)
+    elif remote_type == "collection":
+        ctx.obj = PulpAnsibleCollectionRemoteContext(pulp_ctx)
+    else:
+        raise NotImplementedError()
+
+
+def remote_options(f: Callable[..., RT]) -> Callable[..., RT]:
+    @click.option("--ca-cert", help="A PEM encoded CA cerificate")
+    @click.option("--client-cert", help="A PEM encoded client certificate")
+    @click.option("--client-key", help="A PEM encode private key")
+    @click.option("--tls-validation", help="If True, TLS peer validation must be performed")
+    @click.option("--proxy-url", help="Format: scheme//user:password@host:port")
+    @click.option("--username", help="Used for authentication when syncing")
+    @click.option("--password")
+    @click.option("--download-concurrency", help="Total number of simultaneous connections")
+    @click.option("--policy", help="Policy to use when downloading")
+    @click.option("--total-timeout", help="aiohttp.ClientTimeout.total for download connections")
+    @click.option(
+        "--connect-timeout", help="aiohttp.ClientTimeout.connect for download connections"
+    )
+    @click.option(
+        "--sock-connect-timeout", help="aiohttp.ClientTimeout.sock_connect for download connections"
+    )
+    @click.option(
+        "--sock-read-timeout", help="aiohttp.ClientTimeout.sock_read for download connections"
+    )
+    def config_options(**kwargs: Dict[str, Any]) -> Any:
+        f(**kwargs)
+
+    return config_options
+
+
+remote.add_command(list_entities)
+remote.add_command(show_by_name)
+remote.add_command(destroy_by_name)
+
+
+@remote.command(name="create")
+@click.option("--name", required=True)
+@click.option("--url", required=True)
+@remote_options
+@pass_entity_context
+@pass_pulp_context
+def create_role(
+    pulp_ctx: PulpContext,
+    remote_ctx: PulpEntityContext,
+    name: str,
+    url: str,
+    **remote_options: Dict[str, Any],
+) -> None:
+    """Use '-t collection' after remote to create a collection remote"""
+    body: Dict[str, str] = {"name": name, "url": url}
+    create_remote(
+        pulp_ctx=pulp_ctx, remote_ctx=remote_ctx, body=body, remote_options=remote_options
+    )
+
+
+@remote.command(name="create")
+@click.option("--name", required=True)
+@click.option("--url", required=True)
+@click.option("--requirements-file", help="Collections only: A Collection requirements yaml")
```

Comment:
I would probably have this be a file option. I think in most cases, users will actually have an actual requirements.yaml file they'll want to use. Perhaps have `--requirements` and `--requirements-file` options which take the requirements file contents and an actual requirements.yaml file respectively?

Reply:
I'm ok with just taking a file path at the start. Having both is also ok w/ me.